### ISSUE TITLE
Firefox Can't Download Extension

### ISSUE THREAD

**Issue by zeqwx** [saucettv/VideoAdBlockForTwitch#9]
Hello, for some reason everytime I try to download your extension it keeps on saying "Download failed. Please check your connection." I'm on Firefox with the latest version, and all other extension downloads work. Here's a screenshot, 
![firefox](https://user-images.githubusercontent.com/87872967/126818542-152bdfa6-68d8-4547-aff1-c8c5498eb5ba.PNG)
What do I do?

**Comment by saucettv** [comment]
Hi, I saw this issue before for another user on Firefox. I have no idea why it happens to be honest, as the extension is hosted and distributed by them.

I did leave some troubleshooting info on a previous issue, hopefully it will help you https://github.com/saucettv/VideoAdBlockForTwitch/issues/5

**Comment by zeqwx** [comment]
I tried the 3rd link in that issue https://support.mozilla.org/en-US/kb/unable-install-add-ons-extensions-or-themes?redirectslug=Unable+to+install+add-ons&redirectlocale=en-US#w_corrupt-extension-files and now it shows this, 
![firefox 2](https://user-images.githubusercontent.com/87872967/126901286-f016fd42-9114-4164-88d8-3f3c83929bd5.PNG)
What does this mean? Why does deleting extension.json change the error to "An unexpected error occurred during installation."

**Comment by zeqwx** [comment]
I already tried all the other options btw

**Comment by saucettv** [comment]
Sorry I really don't know, it's only been reported to me twice and it's really a Firefox issue. What happens if you right click the "Add to Firefox" button and click save link as, does it save the file to your computer? If it does, it would rule out any antivirus stuff. 

The extensions are stored in C:\Users\*YourUserName*\AppData\Roaming\Mozilla\Firefox\Profiles\*YourProfileName*\extensions

This extension would have the file name {3385c2d8-dcfd-4f92-adb7-5d8429dee164} check if you see it in that folder or the staged folder.

It might of corrupted half way through install or something, so if you see any files with that file name, you could try deleting them and then closing firefox, re-open it and try adding the add-on again.

**Comment by zeqwx** [comment]
I tried doing that and it just downloads a gif file, and if i click open in a new tab it opens up to a weird image with a dot in the middle, 
![firefox 3](https://user-images.githubusercontent.com/87872967/126901995-9d8bdc35-ef29-4579-bae2-283684028292.PNG)
And there's no file with that name in the folder, here's the screenshot, 
![firefox 4](https://user-images.githubusercontent.com/87872967/126902036-7208ad9b-ccb3-4625-90a8-4b03e0db5eb4.PNG)
Is there really nothing I can do?

**Comment by saucettv** [comment]
That's strange, that's the problem then, when I click right click save as, it actually downloads the extension. The link is https://addons.mozilla.org/firefox/downloads/file/3810893/video_ad_block_for_twitch-4.9.1-fx.xpi

You could try using that link to download the add-on. Or download it via Chrome (If Firefox doens't let you) and install it manually. On the Firefox add-ons page, where it lists your installed add-ons, there is a cog icon, click that, then click "Install add-on from file" and select the downloaded file from earlier.

**Comment by saucettv** [comment]
One thing I also should say is that the extension needs to have Firefox version 57+ for it to work.

**Comment by saucettv** [comment]
If even that doesn't work, it most likely is another extension doing something to stop the install. You can use the trouble shooting mode in Firefox to confirm it or not https://support.mozilla.org/en-US/kb/diagnose-firefox-issues-using-troubleshoot-mode

**Comment by zeqwx** [comment]
Ok, so I tried that link and the same thing happens, and same on chrome, but I tried downloading from my phone and it worked, so I sent it to my computer and now there's no option to "Install add-on from file". Here, 
![firefox 5](https://user-images.githubusercontent.com/87872967/126903172-91f7c4a4-1073-49a6-a239-957faacd5d6b.PNG)

**Comment by zeqwx** [comment]
And I tried troubleshooting mode already, same error

**Comment by zeqwx** [comment]
But basically now I have the file since I downloaded it from my phone but there's no option to install it from file.

**Comment by saucettv** [comment]
Not sure why you don't have that option. https://support.mozilla.org/en-US/kb/find-and-install-add-ons-add-features-to-firefox "For advanced users: You can also install add-ons from a file. If you saved the installer file to your computer, (for example, an .xpi or .jar file) you can install the add-on using the gear icon gear icon in the upper-right area of the Add-on Manager Extensions panel. Select Install Add-on from file... from the menu, then find and select the file."

![1](https://user-images.githubusercontent.com/81865589/126903540-5a05a44d-7246-49e6-927d-5f7450e42f17.png)


It also says you can drag and drop the file onto the add-on manager page. I just tried it myself and it does work with the drag and drop.

Are you using a nightly version of Firefox? Or is it the stable version?

**Comment by zeqwx** [comment]
Maybe it's because of the configs I used from privacytools.io? I'm not sure though, I'm using the normal version of Firefox.

**Comment by zeqwx** [comment]
Here, I used these configs, https://privacytools.io/browsers/#about_config just those, I didn't use the user.js recommended on there.

**Comment by zeqwx** [comment]
Let me try resetting firefox, one second.

**Comment by saucettv** [comment]
It could be, but I don't see anything there that could disable that option. You could try downloading Firefox nightly, temporarily, as it keeps stable installed, so you will have two Firefox versions installed at the same time.

Once the nightly version opens, don't change any settings and go straight to https://addons.mozilla.org/en-US/firefox/addon/video-ad-block-for-twitch and try installing it. If it works, you know it's something in your stable version config.

**Comment by saucettv** [comment]
With Firefox open and focused, you can also try "With Firefox open, hit Ctrl+ o (or Cmd+ o) > browse to the extension > double click the extension (or select Open) > Install Now. "

**Comment by zeqwx** [comment]
Alright, so I refreshed firefox to factory settings, re-inputed my configs from privacytools.io and the option appeared, must have been a bug or something, so I downloaded the xpi file I got from my phone, and it works! Just still don't know why this happened in the first place, but I guess it doesn't matter now, anyways, must've been something weird with my computer since my ISP isn't blocking your add-on, since I could download it fine from my phone. But yeah, it works now, just wondering if it still updates even though I downloaded it manually?

**Comment by saucettv** [comment]
You could now try downloading it normally through the https://addons.mozilla.org/en-US/firefox/addon/video-ad-block-for-twitch page, but the file link I gave you is the signed AMO version from Mozilla, so it should update. Just make sure to remove the old one first.

If you still get trouble, go to https://addons.mozilla.org/en-US/firefox/search/?q=video%20adblock%20for%20twitch , right click the page, click inspect, then click the console tab, keep it open, then click on the video adblock for twitch extension, then click add to firefox. You should then get a read out in the console of any errors and it might have more information.

It might of been a strange setting somewhere, the extension itself doesn't have any strange permissions or anything, so not really sure what happened.

**Comment by zeqwx** [comment]
Still doesn't work, tried removing it and redownloading it and got the same error, the console says something about cookies getting rejected and "Add-on "{3385c2d8-dcfd-4f92-adb7-5d8429dee164}" not found so setting status to UNINSTALLED; exact error: Error: Addon not found"

**Comment by zeqwx** [comment]
And the last message after clicking add to firefox is always "Ignoring promise rejection during installation: Error" in the console.

**Comment by zeqwx** [comment]
I think for some reason the add-on website just can't find the file for me, this is so confusing.

**Comment by saucettv** [comment]
Earlier, you did have an add-on for CookiesAutoDelete, could maybe be something to do with that. Although I think it might be the privacytools config:

"privacy.firstparty.isolate = true
    A result of the Tor Uplift effort, this preference isolates all browser identifier sources (e.g. cookies) to the first party domain, with the goal of preventing tracking across different domains. (Don't do this if you are using the Firefox Addon "Cookie AutoDelete" with Firefox v58 or below.)"

Here is my console output for a successful install.

16:30:22.760 Do Not Track is enabled browser.js:198:17
16:30:22.760 [GA] Do Not Track Enabled; Google Analytics not loaded and tracking disabled undefined browser.js:198:17
16:30:22.861 adding event listener for "onDisabled" browser.js:198:17
16:30:22.862 adding event listener for "onEnabled" browser.js:198:17
16:30:22.862 adding event listener for "onInstalling" browser.js:198:17
16:30:22.862 adding event listener for "onInstalled" browser.js:198:17
16:30:22.862 adding event listener for "onUninstalling" browser.js:198:17
16:30:22.862 adding event listener for "onUninstalled" browser.js:198:17
16:30:22.862 adding event listener for "onEnabling" browser.js:198:17
16:30:22.862 adding event listener for "onDisabling" browser.js:198:17
16:30:22.862 Global change event listeners have been initialized browser.js:198:17
16:30:32.657 The resource at “https://addons-amo.cdn.mozilla.net/static/Inter-roman-subset-en_de_fr_ru_es_pt_pl_it.var.2ce5ad921c3602b1e5370b3c86033681.woff2” preloaded with link preload was not used within a few seconds. Make sure all attributes of the preload tag are set correctly. search
16:31:26.648 Updating add-on status browser.js:198:17
16:31:26.648 Setting add-on status browser.js:198:17
16:31:26.724 Add-on "{3385c2d8-dcfd-4f92-adb7-5d8429dee164}" not found so setting status to UNINSTALLED; exact error: Error: Addon not found browser.js:198:17
16:31:27.155 [GA: OFF] sendEvent:  
Object { eventAction: "recommended_fallback-no_results", eventCategory: "AMO Addon / Recommendations Shown", eventLabel: "{3385c2d8-dcfd-4f92-adb7-5d8429dee164}", eventValue: undefined, hitType: "event" }
browser.js:198:17
16:31:31.499 [GA: OFF] sendEvent:  
Object { eventAction: "addon", eventCategory: "AMO Addon Installs Started", eventLabel: "{3385c2d8-dcfd-4f92-adb7-5d8429dee164}", eventValue: undefined, hitType: "event" }
browser.js:198:17
16:31:31.515 [install] Adding listener for onDownloadStarted browser.js:198:17
16:31:31.516 [install] Adding listener for onDownloadProgress browser.js:198:17
16:31:31.516 [install] Adding listener for onDownloadEnded browser.js:198:17
16:31:31.516 [install] Adding listener for onDownloadCancelled browser.js:198:17
16:31:31.516 [install] Adding listener for onDownloadFailed browser.js:198:17
16:31:31.517 [install] Adding listener for onInstallStarted browser.js:198:17
16:31:31.517 [install] Adding listener for onInstallProgress browser.js:198:17
16:31:31.517 [install] Adding listener for onInstallEnded browser.js:198:17
16:31:31.517 [install] Adding listener for onInstallCancelled browser.js:198:17
16:31:31.517 [install] Adding listener for onInstallFailed browser.js:198:17
16:31:31.517 Events to handle the installation initialized. browser.js:198:17
16:31:33.565 Event received:  
Object { type: "onInstalling", id: "{3385c2d8-dcfd-4f92-adb7-5d8429dee164}", needsRestart: undefined }...
16:31:33.581 Event received:  
Object { type: "onInstalled", id: "{3385c2d8-dcfd-4f92-adb7-5d8429dee164}", needsRestart: undefined }...
16:31:33.581 [GA: OFF] sendEvent:  
Object { eventAction: "addon", eventCategory: "AMO Addon Installs", eventLabel: "{3385c2d8-dcfd-4f92-adb7-5d8429dee164}", eventValue: undefined, hitType: "event" }...

If you have time might be worth doing a refresh again, don't input the privacytools config, or change any settings, add any add-ons until you try downloading the extension.

**Comment by zeqwx** [comment]
So I basically already tried this, by creating a new profile, and I really don't think the cookies are the problem beacuse in the new profile, the console says this, 

[GA] Google Analytics is enabled undefined browser.js:198:17
[GA: ON] trackingSendWebVitals is enabled undefined browser.js:198:17
[GA: ON] sendEvent:  
Object { eventAction: "notInExperiment", eventCategory: "AMO Experiment Enrollment - 20210531_amo_download_funnel_experiment", eventLabel: undefined, eventValue: undefined, hitType: "event" }
browser.js:198:17
Setting add-on status browser.js:198:17
[GA: ON] set:  
Object { dimension: "dimension4", value: "TAAR_COHORT_INCLUDED" }
browser.js:198:17
[GA: ON] set:  
Object { dimension: "dimension5", value: "TAAR-LITE-AB" }
browser.js:198:17
adding event listener for "onDisabled" browser.js:198:17
adding event listener for "onEnabled" browser.js:198:17
adding event listener for "onInstalling" browser.js:198:17
adding event listener for "onInstalled" browser.js:198:17
adding event listener for "onUninstalling" browser.js:198:17
adding event listener for "onUninstalled" browser.js:198:17
adding event listener for "onEnabling" browser.js:198:17
adding event listener for "onDisabling" browser.js:198:17
Global change event listeners have been initialized browser.js:198:17
Add-on "{3385c2d8-dcfd-4f92-adb7-5d8429dee164}" not found so setting status to UNINSTALLED; exact error: Error: Addon not found browser.js:198:17
[GA: ON] sendEvent:  
Object { eventAction: "recommended_fallback-no_results", eventCategory: "AMO Addon / Recommendations Shown", eventLabel: "{3385c2d8-dcfd-4f92-adb7-5d8429dee164}", eventValue: undefined, hitType: "event" }
browser.js:198:17
Some cookies are misusing the recommended “SameSite“ attribute 3
[GA: ON] sendWebVitalStats:  
Object { name: "FID", delta: 13, id: "v2-1627228587324-9454376220917" }
browser.js:198:17
[GA: ON] sendEvent:  
Object { eventAction: "addon", eventCategory: "AMO Addon Installs Started", eventLabel: "{3385c2d8-dcfd-4f92-adb7-5d8429dee164}", eventValue: undefined, hitType: "event" }
browser.js:198:17
[install] Adding listener for onDownloadStarted browser.js:198:17
[install] Adding listener for onDownloadProgress browser.js:198:17
[install] Adding listener for onDownloadEnded browser.js:198:17
[install] Adding listener for onDownloadCancelled browser.js:198:17
[install] Adding listener for onDownloadFailed browser.js:198:17
[install] Adding listener for onInstallStarted browser.js:198:17
[install] Adding listener for onInstallProgress browser.js:198:17
[install] Adding listener for onInstallEnded browser.js:198:17
[install] Adding listener for onInstallCancelled browser.js:198:17
[install] Adding listener for onInstallFailed browser.js:198:17
Events to handle the installation initialized. browser.js:198:17
[GA: ON] sendEvent:  
Object { eventAction: "addon", eventCategory: "AMO Addon Installs Download Failed", eventLabel: "{3385c2d8-dcfd-4f92-adb7-5d8429dee164}", eventValue: undefined, hitType: "event" }
browser.js:198:17
Ignoring promise rejection during installation: Error

I think the part where it says "Add-on "{3385c2d8-dcfd-4f92-adb7-5d8429dee164}" not found so setting status to UNINSTALLED; exact error: Error: Addon not found browser.js:198:17" after the global line is the issue, for some reason my firefox can't find the file. Since for you it downloads the file and links it.

**Comment by saucettv** [comment]
The "Add-on "{3385c2d8-dcfd-4f92-adb7-5d8429dee164}" not found so setting status to UNINSTALLED; exact error: Error: Addon not found browser.js:198:17" is Firefox looking for the add-on on your PC I think. I had the same message.

You do still have "Some cookies are misusing the recommended “SameSite“ attribute 3" in the console, which I didn't have. Is it possible you used an autoconfig file for Firefox? As that would apply the config to all profiles. 

I'm also not sure what "AMO Experiment Enrollment - 20210531_amo_download_funnel_experiment" is.

**Comment by zeqwx** [comment]
Alright, I gotta sleep now, but is it okay if we can keep this issue open and I try completely uninstalling and reinstalling firefox tomorrow? I'll give you a reply then. That okay?

**Comment by saucettv** [comment]
Yeah no worries, hope you can work it out :)

**Comment by zeqwx** [comment]
Hi again, so I tried exactly that, and I still get the same error, here's the console message, 

[GA] Google Analytics is enabled undefined browser.js:198:17
[GA: ON] trackingSendWebVitals is enabled undefined browser.js:198:17
[GA: ON] sendEvent:  
Object { eventAction: "notInExperiment", eventCategory: "AMO Experiment Enrollment - 20210531_amo_download_funnel_experiment", eventLabel: undefined, eventValue: undefined, hitType: "event" }
browser.js:198:17
Setting add-on status browser.js:198:17
[GA: ON] set:  
Object { dimension: "dimension4", value: "TAAR_COHORT_INCLUDED" }
browser.js:198:17
[GA: ON] set:  
Object { dimension: "dimension5", value: "TAAR-LITE-AB" }
browser.js:198:17
adding event listener for "onDisabled" browser.js:198:17
adding event listener for "onEnabled" browser.js:198:17
adding event listener for "onInstalling" browser.js:198:17
adding event listener for "onInstalled" browser.js:198:17
adding event listener for "onUninstalling" browser.js:198:17
adding event listener for "onUninstalled" browser.js:198:17
adding event listener for "onEnabling" browser.js:198:17
adding event listener for "onDisabling" browser.js:198:17
Global change event listeners have been initialized browser.js:198:17
Add-on "{3385c2d8-dcfd-4f92-adb7-5d8429dee164}" not found so setting status to UNINSTALLED; exact error: Error: Addon not found browser.js:198:17
[GA: ON] sendEvent:  
Object { eventAction: "recommended_fallback-no_results", eventCategory: "AMO Addon / Recommendations Shown", eventLabel: "{3385c2d8-dcfd-4f92-adb7-5d8429dee164}", eventValue: undefined, hitType: "event" }
browser.js:198:17
Some cookies are misusing the recommended “SameSite“ attribute 3
[GA: ON] sendWebVitalStats:  
Object { name: "FID", delta: 15, id: "v2-1627257905377-5941268121229" }
browser.js:198:17
[GA: ON] sendEvent:  
Object { eventAction: "addon", eventCategory: "AMO Addon Installs Started", eventLabel: "{3385c2d8-dcfd-4f92-adb7-5d8429dee164}", eventValue: undefined, hitType: "event" }
browser.js:198:17
[install] Adding listener for onDownloadStarted browser.js:198:17
[install] Adding listener for onDownloadProgress browser.js:198:17
[install] Adding listener for onDownloadEnded browser.js:198:17
[install] Adding listener for onDownloadCancelled browser.js:198:17
[install] Adding listener for onDownloadFailed browser.js:198:17
[install] Adding listener for onInstallStarted browser.js:198:17
[install] Adding listener for onInstallProgress browser.js:198:17
[install] Adding listener for onInstallEnded browser.js:198:17
[install] Adding listener for onInstallCancelled browser.js:198:17
[install] Adding listener for onInstallFailed browser.js:198:17
Events to handle the installation initialized. browser.js:198:17
[GA: ON] sendEvent:  
Object { eventAction: "addon", eventCategory: "AMO Addon Installs Download Failed", eventLabel: "{3385c2d8-dcfd-4f92-adb7-5d8429dee164}", eventValue: undefined, hitType: "event" }
browser.js:198:17
Ignoring promise rejection during installation: Error

I'm so confused on why this is happening

**Comment by saucettv** [comment]
Earlier when you tried downloading the extension via Chrome, using the link https://addons.mozilla.org/firefox/downloads/file/3810893/video_ad_block_for_twitch-4.9.1-fx.xpi , did you mean the download failed? Or that you got the same GIF type picture?

If it didn't download the xpi file in Chrome, then it sounds like something on your PC is blocking the download. I don't know your setup, but could be a whole host of things like VPN, custom DNS, Firewall, Antivirus etc.

**Comment by saucettv** [comment]
Could check it by installing a new fresh nightly version of Firefox https://www.mozilla.org/en-US/firefox/channel/desktop/ don't configure anything in it, just go straight to the add-on download page and try downloading it from there. If it does the same thing, it's something on your PC blocking it.

**Comment by zeqwx** [comment]
Yeah it doesn't work on chrome, it shows a gif type picture, the same as firefox nightly version, so it must be something on my PC. I tried disabling my antivirus but it still doesn't work, I was planning to factory reset my PC anyway so I'll tell you how that goes. I'll install firefox first before my antivirus and try it again after installing it to see if that's the issue.

**Comment by saucettv** [comment]
Hi, did you have time to do your reset? How did it go?

**Comment by zeqwx** [comment]
Hi again, sorry I've been really busy the past week, I'll do the reset today and tell you how it goes.

**Comment by saucettv** [comment]
No problem, take your time, no rush at all, was just checking in on you :)

**Comment by zeqwx** [comment]
Hey again, so I did a full clean reinstall of windows and the first time it didn't work, but I'm not sure if that's because I installed my antivirus first or logged into firefox first so I could sync. So I did a clean reinstall again and this time I made sure to install firefox and your add on first. And it worked! But I still can't figure out why it worked. So I installed the antivirus after and removed the add on and reinstalled it and it worked again! And the link when you click "open in new tab" actually installs the add on file. Then I thought the issue must've been logging into firefox, so I tried logging into firefox again to sync and it worked after that too! I'm still confused as I thought it was my antivirus or firefox logging in that was causing the issue since it persisted after the reinstall, but now I'm just confused as it still works after doing both. At least it works now but I'm still scratching my head about this. Any thoughts?
